src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -235,6 +239,15 @@ private void processZipFile(
         }
     }
 
+    private ZipInputStream loadIntoMemory(BlobInputStream stream) {
+        try {
+            byte[] array = toByteArray(stream);
+            return new ZipInputStream(new ByteArrayInputStream(array));
+        } catch (IOException exception) {
+            throw new ZipFileLoadException("Error loading zip archive", exception);
```

Comment:
Suggestion (minor): The message could mention blob instead of zip, as we're downloading a blob. At this point it's unknown if it's a valid zip or not. The exception could also have `Blob` in its name, instead of `Zip`.